community/index/src/main/java/org/neo4j/index/internal/gbptree/GBPTree.java

```diff
@@ -236,13 +238,7 @@ default void startupState( boolean clean )
      * on stable storage, no writes are allowed to happen.
      * For this reason both writer and check-pointing acquires this lock.
      */
-    private final Lock writerCheckpointMutex = new ReentrantLock();
-
-    /**
-     * Currently an index only supports one concurrent writer and so this boolean will act as
-     * guard so that only one thread can have it at any given time.
-     */
-    private final AtomicBoolean writerTaken = new AtomicBoolean();
+    private final ReadWriteLock writerCheckpointMutex = new StampedLock().asReadWriteLock();
```

Comment:
now quite sure, I understand what you mean. Let me be more specific: does this lock needs to be StampedLock or it can be `ReentrantReadWriteLock` as well?